File: Task/Compare-a-list-of-strings/Go/compare-a-list-of-strings.go
package cmp

func AllEqual(strings []string) bool {
	for _, s := range strings {
		if s != strings[0] {
			return false
		}
	}
	return true
}

func AllLessThan(strings []string) bool {
	for i := 1; i < len(strings); i++ {
		if !(strings[i - 1] < s) {
			return false
		}
	}
	return true
}

File: Task/Compare-a-list-of-strings/Erlang/compare-a-list-of-strings.erl
-module(compare_strings).

-export([all_equal/1,all_incr/1]).

all_equal(Strings) ->
	all_fulfill(fun(S1,S2) -> S1 == S2 end,Strings).

all_incr(Strings) ->
	all_fulfill(fun(S1,S2) -> S1 < S2 end,Strings).

all_fulfill(Fun,Strings) ->
	lists:all(fun(X) -> X end,lists:zipwith(Fun, lists:droplast(Strings), tl(Strings)) ).
